src/jemalloc.c

```diff
@@ -4148,8 +4170,41 @@ batch_alloc(void **ptrs, size_t num, size_t size, int flags) {
 					bin = &tcache->bins[ind];
 				}
 			}
+			/*
+			 * If we don't have a tcache bin, we don't want to
+			 * immediately give up, because there's the possibility
```

Comment:
This (i.e. that the user specified MALLOCX_TCACHE_NONE) is the only possibility, right?

Reply:
As far as I understand, that should be the main one. Some other possibilities include:
- `tcache` has not been initialized, which should be rare.
- `tcache` is disabled by the caller explicitly.